src/cmd/linuxkit/run_azure.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+// This program requires that the following environment vars are set:
+
+// AZURE_TENANT_ID: contains your Azure Active Directory tenant ID or domain
+// AZURE_SUBSCRIPTION_ID: contains your Azure Subscription ID
+// AZURE_CLIENT_ID: contains your Azure Active Directory Application Client ID
+// AZURE_CLIENT_SECRET: contains your Azure Active Directory Application Secret
+
+func runAzure(args []string) {
+	flags := flag.NewFlagSet("azure", flag.ExitOnError)
+	invoked := filepath.Base(os.Args[0])
+	flags.Usage = func() {
+		fmt.Printf("USAGE: %s run azure [options] [name]\n\n", invoked)
+		fmt.Printf("'imagePath' specifies a VHD file which will be\n")
+		fmt.Printf("uploaded before it is run.\n\n")
+		fmt.Printf("Options:\n\n")
+		flags.PrintDefaults()
+	}
+
+	resourceGroupName := flags.String("resourceGroupName", "", "Name of resource group to be used for VM")
+	location := flags.String("location", "westus", "Location of the VM")
+	accountName := flags.String("accountName", "linuxkitstorage", "Name of the storage account")
+
+	subscriptionID := getEnvVarOrExit("AZURE_SUBSCRIPTION_ID")
+	tenantID := getEnvVarOrExit("AZURE_TENANT_ID")
+
+	clientID := getEnvVarOrExit("AZURE_CLIENT_ID")
+	clientSecret := getEnvVarOrExit("AZURE_CLIENT_SECRET")
+
+	if err := flags.Parse(args); err != nil {
+		log.Fatal("Unable to parse args")
+	}
+
+	remArgs := flags.Args()
+	if len(remArgs) == 0 {
+		fmt.Printf("Please specify the image to run\n")
+		flags.Usage()
+		os.Exit(1)
+	}
+	imagePath := remArgs[0]
+
+	rand.Seed(time.Now().UTC().UnixNano())
+	virtualNetworkName := fmt.Sprintf("linuxkitvirtualnetwork%d", rand.Intn(1000))
+	subnetName := fmt.Sprintf("linuxkitsubnet%d", rand.Intn(1000))
+	publicIPAddressName := fmt.Sprintf("publicip%d", rand.Intn(1000))
+	networkInterfaceName := fmt.Sprintf("networkinterface%d", rand.Intn(1000))
+	virtualMachineName := fmt.Sprintf("linuxkitvm%d", rand.Intn(1000))
+
```

Comment:
I don't know Azure well enough, so I'm not sure if hardcoding `linuxkit` here is a good idea.